internal/datastore/memdb/readonly.go

```diff
@@ -0,0 +1,353 @@
+package memdb
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/hashicorp/go-memdb"
+	"github.com/jzelinskie/stringz"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/authzed/spicedb/internal/datastore/options"
+	"github.com/authzed/spicedb/pkg/datastore"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+type txFactory func() (*memdb.Txn, error)
+
+type memdbReader struct {
+	TryLocker
+	txSource txFactory
+	revision datastore.Revision
+	initErr  error
+}
+
+// QueryRelationships reads relationships starting from the resource side.
+func (rot *memdbReader) QueryRelationships(
```

Comment:
what is rot? i assumed read only transaction, but that's not accurate as this is a reader which has a tx factory and isn't an active transaction itself.